File: aws-node-rest-api-with-dynamodb/todos/list.js
'use strict';

const AWS = require('aws-sdk');

const dynamoDb = new AWS.DynamoDB.DocumentClient();
const params = {
  TableName: 'todos',
};

module.exports.list = (event, context, callback) => {
  // fetch all todos from the database
  dynamoDb.scan(params, (error, result) => {
    // handle potential errors
    if (error) {
      console.error(error); // eslint-disable-line no-console
      callback(new Error('Couldn\'t fetch the todos.'));
      return;
    }

    // create a resonse
    const response = {
      statusCode: 200,
      body: JSON.stringify(result.Items),
    };
    callback(null, response);
  });
};

File: aws-python-pynamodb-s3-sigurl/asset/list.py
import http.client as httplib
from asset.asset_model import AssetModel
from log_cfg import logger


def asset_list(event, context):
    logger.debug('event: {}, context: {}'.format(event, context))

    results = AssetModel.scan()
    return {
        'statusCode': httplib.OK,
        'body': {
            'items': [dict(result) for result in results]
        }
    }
